source/class/qx/Promise.js

```diff
@@ -322,13 +333,13 @@ qx.Class.define("qx.Promise", {
      * has items in random order, it means that .map can be used for concurrency coordination unlike
      * .all.
      *
-     * @param iterable {Iterable} An iterable object, such as an Array
      * @param iterator {Function} the callback, with <code>(value, index, length)</code>
-     * @param options {Object?} options; can be:
-     *  <code>concurrency</code> max nuber of simultaneous filters, default is <code>Infinity</code>
+     * @param options {Object?} Either:
+     *   A native object with one key: <code>concurrency</code>: max number of simultaneous filters, default is <code>Infinity</code>
+     * Or: any other object (subclass of qx.core.Object), in which case this will be the context for the iterator
      * @return {qx.Promise}
      */
-    map(iterable, iterator, options) {
+    map(iterator, options) {
```

Comment:
It looks like there are lots of these. I won't point out each one, but all of these methods must be BC if not a major version bump... and even if it is a major version bump, this could mean major changes to users' apps.

Reply:
i think that this deserves a proper check over on all the methods, but AIUI none of the member methods need an iterable passed in

Reply:
They may not require an iterable to be passed in, but the argument was there so any users of `qx.Promise` having seen that would have passed _something_ as that iterable; the iterator would be the second argument they passed, not the first. Am I missing something?

Reply:
if they did, then tjheir code crashes because a function (ie `iterator`) is expected as the first argument.  I just tried this code in the playground:

```
let p = new qx.Promise(resolve => resolve([1,2,3]));
p.forEach(x => console.log(x));
```
which outputs:
```
1
2
3
```

then:

```
let p = new qx.Promise(resolve => resolve([1,2,3]));
p.forEach([4,5,6], x => console.log(x));
```
which outputs:
```
qx.Promise: Unhandled promise rejection: TypeError: expecting a function but got [object Array]
```

Reply:
I don't understand, but empirical evidence seems adequate proof.